keps/sig-node/753-sidecar-containers/README.md

```diff
@@ -192,584 +173,1331 @@ logs are lost.
+sequencing by using blocking nature of `postStart` hook. So using the sidecar
+containers instead is not adding any benefits.
+
+As for injecting sidecars into Pods with regular containers run as sidecars -
+webhooks will likely inject sidecar containers to be first, so the risk is
+minimal here as well.
+
+#### Scenario 2. Balloon sidecars
+
+Users may decide to start the “large request” sidecar containers early to
+pre-allocate resources for other containers. The same time asking for less
+resources for a regular containers hoping to reuse what is allocated by the
+“baloon” sidecar. This was previously impossible to pre-allocate resources like
+CPU before any init containers run and may be more critical when resource
+managers like CPU or topology managers are being used. If this pattern have any
+benefits, people may be incentivized to abuse sidecars this way. However it’s
+unclear if this has any benefits for users at all. 
+
+#### Scenario 3. Long initialization tasks running in-parallel
+
+Users may decide to implement long initialization tasks that will run
+in-parallel with other initialization tasks. Users may also decide to run new
+type of initialization tasks  like image preloading for workload containers from
+the sidecar container, which will make it run in parallel with the
+Initialization tasks. This is impossible to do today as only a single Init
+container is run at a time today. For containers with lengthy initialization
+this pattern may be abused and can lead to the pattern of sidecars
+synchronization when the first workload container will wait for all Init sidecar
+containers to complete. This pattern may lead to race conditions and be error
+prone.
+
+Today similar behavior can be achieved by running initialization tasks as
+regular containers, with the special container that blocks workload from
+execution using synchronization logic in the `PostStart` handler. Sidecars support
+makes implementation of this feature easier.
+
+There is not much risk, however, even if the user abuses this pattern by
+converting all init containers in sidecars. Whenever this pattern is useful, the
+user will most likely need to spend time understanding the logic of different
+Init Containers and making sure it is not conflicting. This pattern also likely
+prohibited for large scale customers because sidecar-based initialization will
+cost resources even after initialization is complete.
+
+#### Scenario 4. Sidecar that never becomes ready 
+
+This is the failure mode when the sidecar never becomes ready. This is
+functionally equivalent to the init container that never completes and not
+allowing users to implement any other ways to abuse kubelet.
+
+#### Scenario 5. Intentional failing or terminating sidecars
+
+Users may implement a sidecar container that intentionally crashes or terminates
+often. This scenario functionally similar to a regular container restarts often
+so not much additional overhead or side effects will be created. 
```

Comment:
Not sure 4/5 are worth documenting but fine either ways.

Reply:
yes, they are here for completeness